src/poly/commitment/msm.rs

```diff
@@ -34,52 +34,58 @@ impl<'a, C: CurveAffine> MSM<'a, C> {
         self.other_bases.extend(other.other_bases.iter());
 
         if let Some(g_scalars) = &other.g_scalars {
-            self.add_to_g(&g_scalars);
+            self.add_to_g_scalars(&g_scalars);
         }
 
         if let Some(h_scalar) = &other.h_scalar {
-            self.add_to_h(*h_scalar);
+            self.add_to_h_scalar(*h_scalar);
         }
     }
 
     /// Add arbitrary term (the scalar and the point)
-    pub fn add_term(&mut self, scalar: C::Scalar, point: C) {
+    pub fn append_term(&mut self, scalar: C::Scalar, point: C) {
         self.other_scalars.push(scalar);
         self.other_bases.push(point);
     }
 
     /// Add a vector of scalars to `g_scalars`. This function will panic if the
     /// caller provides a slice of scalars that is not of length `params.n`.
-    // TODO: parallelize
-    pub fn add_to_g(&mut self, scalars: &[C::Scalar]) {
+    pub fn add_to_g_scalars(&mut self, scalars: &[C::Scalar]) {
         assert_eq!(scalars.len(), self.params.n as usize);
         if let Some(g_scalars) = &mut self.g_scalars {
-            for (g_scalar, scalar) in g_scalars.iter_mut().zip(scalars.iter()) {
-                *g_scalar += scalar;
-            }
+            parallelize(g_scalars, |g_scalars, start| {
+                for (g_scalar, scalar) in g_scalars.iter_mut().zip(scalars[start..].iter()) {
+                    *g_scalar += scalar;
+                }
+            })
         } else {
             self.g_scalars = Some(scalars.to_vec());
         }
     }
 
-    /// Add term to h
-    pub fn add_to_h(&mut self, scalar: C::Scalar) {
+    /// Add to `h_scalar`
+    pub fn add_to_h_scalar(&mut self, scalar: C::Scalar) {
```

Comment:
`append_to_h_scalar`

Reply:
this does actually add to the scalar; the confusingly named method was `add_term()` which we've changed in this PR to `append_term()`.